File: Solution in CPP/Que4.cpp
// Implement a function that checks whether a given number is prime or not.

#include<bits/stdc++.h>
using namespace std;

bool isPrime(int num){
    if(num <= 1){
        return false;
    }


    for(int i = 2; i*i<= num; i++){
        if(num%i == 0){
            return false;
        }
    }

    return true;
}

int main(){
    int n = 11;
    if(isPrime(n)){
        cout<<"Prime Number"<<endl;
    }

    else{
        cout<<"Non Prime Number"<<endl;
    }

    return 0;
}

File: Solution in RUST/Que4.rs
// Implement a function that checks whether a given number is prime or not.


fn is_prime(num: u32) -> bool {
    if num <= 1 {
        return false;
    }

    let mut i = 2;
    while i * i <= num {
        if num % i == 0 {
            return false;
        }
        i += 1;
    }

    true
}

fn main() {
    let n = 11;
    if is_prime(n) {
        println!("Prime Number");
    } else {
        println!("Non Prime Number");
    }
}
